tests/test.py

```diff
@@ -924,3 +931,89 @@ def test_01_cli_version(self):
         self.assertIn("libqpdf", stdout)
         self.assertIn("OS-", stdout)
         self.assertIn("OS_Version-", stdout)
+
+
+class TestBatch10(PdfArrangerTest):
+    """Test show-save-warnings config settings"""
+
+    def test_01_default(self):
+        """Test default setting"""
+        self._start(["tests/test_encrypted.pdf"])
+        dialog = self._app().child(roleName="dialog")
+        passfield = dialog.child(roleName="password text")
+        passfield.text = "foobar"
+        dialog.child(name="OK").click()
+        self._wait_cond(lambda: dialog.dead)
+
+        self._mainmenu("Preferences")
+        dialog = self._app().child(roleName="dialog")
+        savepanel_name = "Saving/exporting to single file"
+        if have_pikepdf8():
+            savepanel = dialog.child(name=savepanel_name)
+            checkboxes = self._find_by_role("check box", savepanel)
+            self.assertTrue(checkboxes[0].checked)
+        else:
+            panels = self._find_by_role("panel", dialog)
+            self.assertNotIn(savepanel_name, [p.name for p in panels])
+        dialog.child(name="OK").click()
+        self._wait_cond(lambda: dialog.dead)
+
+        self._mainmenu("Save As…")
+        self._save_as_chooser("encrypted.pdf", ["encrypted.pdf"] )
+        with pikepdf.open(os.path.join(self.__class__.tmp, "encrypted.pdf"), password="foobar"):
```

Comment:
Why is this "with pikepdf.." here, is it a mistake?

Reply:
Yes - another good find by you :). Lines 963 and 964 should be deleted. Will fix it.